Sources/SourceControl/GitRepository.swift

```diff
@@ -1161,8 +1161,16 @@ extension GitFileSystemView: @unchecked Sendable {}
 
 // MARK: - Errors
 
-private struct GitShellError: Error {
+public struct GitShellError: Error, CustomStringConvertible {
```

Comment:
I'd rather proceed cautiously and make this `package` or `@_spi(SwiftPMTestSuite)` first, unless it's shown that making it `public` is absolutely necessary.

Reply:
Package it is, [4dc1860](https://github.com/swiftlang/swift-package-manager/pull/8152/commits/4dc1860fb7f78ae3b59a4481dbb27a88af89e6a0).